dashboard/app/models/concerns/user/assigned_courses_and_scripts.rb

```diff
@@ -0,0 +1,245 @@
+module User::AssignedCoursesAndScripts
+  extend ActiveSupport::Concern
+  include User::SectionParticipation
+
+  # Returns an array of hashes storing data for each unique course assigned to # sections that this user is a part of.
+  # @return [Array{CourseData}]
+  def assigned_courses
+    section_courses.map(&:summarize_short)
+  end
+
+  def assigned_course?(course)
+    section_courses.include?(course)
+  end
+
+  # Returns the set of courses the user has been assigned to or has progress in.
+  def courses_as_participant
+    visible_scripts.filter_map(&:unit_group).concat(section_courses).uniq
+  end
+
+  def visible_scripts
+    scripts.map(&:cached).select {|s| [Curriculum::SharedCourseConstants::PUBLISHED_STATE.stable, Curriculum::SharedCourseConstants::PUBLISHED_STATE.preview].include?(s.get_published_state)}
+  end
+
+  def assigned_script?(script)
+    section_scripts.include?(script) || section_courses.include?(script&.unit_group)
+  end
+
+  # Checks if there are any launched scripts assigned to the user.
+  # @return [Array] of Scripts
+  def visible_assigned_scripts
+    user_scripts.where("assigned_at").
+      map {|user_script| Unit.where(id: user_script.script.id).select(&:launched?)}.
+      flatten
+  end
+
+  # Checks if there are any launched scripts assigned to the user.
+  # @return [Boolean]
+  def any_visible_assigned_scripts?
+    visible_assigned_scripts.any?
+  end
+
+  # Checks if there are any launched scripts or courses assigned to the user.
+  # @return [Boolean]
+  def assigned_course_or_script?
+    assigned_courses.any? || any_visible_assigned_scripts?
+  end
+
+  # Query to get the user_script the user was most recently assigned.
+  def most_recently_assigned_user_script
+    user_scripts.
+      where.not(assigned_at: nil).
+      reorder(assigned_at: :desc).
+      first
+  end
+
+  # Get the UnitGroupUnit for the most recently assigned UserScript.
+  def most_recently_assigned_unit_group_unit
+    unit = most_recently_assigned_user_script&.script
+    return unless unit
+    # UserScript doesn't record the UnitGroup the user was in, so we will assume
+    # it is the most recently created section.
+    section = sections_as_student.select {|s| !s.hidden && s.script_id == unit.id}.last
+    Queries::Courses.unit_group_unit(unit, section&.unit_group)
+  end
+
+  # Get script object of the user_script the user was most recently
+  # assigned.
+  def most_recently_assigned_script
+    most_recently_assigned_user_script&.script
```

Comment:
Added the Safe Navigation Operator (`&`) here, which was missing in the old method.